UPGRADE-3.x.md

```diff
@@ -0,0 +1,8 @@
+UPGRADE 3.x
+===========
+
+### Tests
+
+All files under the ``Tests`` directory are now correctly handled as internal test classes. 
```

Comment:
Argh! Didn't spot that. A new PR should be created from a cherry-pick of this, and then a forward merge should apply cleanly, no need for a revert IMO. Sorry for the mess!